flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -219,6 +250,20 @@ def index(self):
   def metadata(self):
     return self._metadata
 
+  @classmethod
+  def from_variable(cls, variable: Variable[tp.Any], index: int) -> VariableDef:
+    metadata = vars(variable).copy()
+    del metadata['raw_value']
+    del metadata['_trace_state']
+    return cls(type(variable), index, metadata)
+
+  def to_variable(self, value: Node) -> Variable[Node]:
+    variables = object.__new__(self._type)
```

Comment:
Add a comment here explaining why you are calling `object.__new__` and not `self._type(...)`.